content/blog/2020-03-01-bookdown-learnings/index.Rmd

```diff
@@ -0,0 +1,257 @@
+
+#### 2. How to work in RStudio and GitHub to collaborate on a project
+
+I'm an R beginner. 
+My blog posts have never contained any R code. 
+I've always written them in Markdown using the Atom editor and I've always run git from the terminal, building on recipes that Scott Chamberlain and Karthik Ram have shared with me. 
+After three years working in git and GitHub, I'm an intermediate user who still lacks robust mental model and the vocabulary to explain it[^2]. 
+(This is not self-deprecation.) 
+This bookdown Blog Guide project was the first time I used RStudio hooked up to GitHub to collaborate. 
+In the process, I have become much more comfortable with installing packages as needed and keeping R, RStudio, and package versions up to date with a weekly calendar reminder.
+
+- For setting up RStudio and knowing how to open a project from .Rproj I loved R-Ladies Sydney's "[opinionated tour of RStudio](https://rladiessydney.org/courses/ryouwithme/01-basicbasics-1/)". 
+- I used happygitwithR "[Connect RStudio to Git and GitHub](https://happygitwithr.com/rstudio-git-github.html)" to do exactly that. 
+Years ago I had already used [happygitwithR](https://happygitwithr.com/) to set up my git/GitHub from square-one and my https credentials.
+- Maëlle and I both worked on feature branches of master. 
+On any given day we each might work on mulitple chapters of the book. Because each book chapter is generated from one Rmd file (more on that in tip #4), and you don't want one pull request to be _too_ complex, I would create a separate feature branch for any chapter or file I worked on. I would periodically push my work to GitHub in a pull request (more in tip #3) with "_Draft_" status so Maëlle could see I was not yet seeking her review. At the end of a day's commits, I would update the pull request status to "_Ready for review_" and assign Maëlle as reviewer. She would review, edit, and merge my pull requests and open new branches of her own for me to review.
+
+#### 3. How to usethis
+
+At the start of the project, Maëlle asked me if I used the usethis package. 
+I did not. 
+But I bought into it because "[usethis](https://usethis.r-lib.org/) is a workflow package: it automates repetitive tasks that arise during project setup and development, both for R packages and non-package projects."
+The trickiest part, for me, was commiting to setting up and forcing myself to use usethis pull request helpers despite not really understanding how it would help me. 
+
+From previous work I had already followed happygitwithr to set up my [GitHub authentication via https](https://happygitwithr.com/credential-caching.html) and a [GitHub Personal Access Token (PAT)](https://happygitwithr.com/github-pat.html) in my `.Renviron`.
+I followed usethis [setup instructions](https://usethis.r-lib.org/articles/articles/usethis-setup.html) to edit my `.Rprofile` so that its functions would be available without explicitly calling the package.
```

Comment:
Fun fact (not to add in the post), I suggested this usethis article :sunglasses: https://www.tidyverse.org/blog/2018/02/usethis-1-3-0/#acknowledgements

Reply:
coooool!
oh, and the real reason I bought into using usethis, is because you strongly suggested I should :-)